### PR TITLE
chore: Add last function to util

### PR DESCRIPTION
**Pull request by marikaner**
## Context

We are slowly getting rid of rambda. `last` can be quite useful, so I added it to our util package.

## Definition of Done

Please consider all items and remove only if not applicable.

- [x] Tests created/adjusted for your changes.
- [ ] Release notes updated.
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".
- [ ] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)
- [ ] If applicable: Properly documented (JSDoc of public API)
- [ ] If applicable: Check if `node run doc` still works.

### CODE REVIEW

File: packages/core/src/odata/v2/odata-batch-request-builder.ts [hunk 1]
```diff
@@ -252,7 +251,7 @@ function partitionBatchResponse(response: string, lineBreak: string): string[] {
   }
 
   // E.g., --batch_1234
-  const partSeparator = head(response.split(lineBreak, 1));
+  const partSeparator = response.split(lineBreak, 1)[0];
```
Comment (FrankEssenberger) [inline packages/core/src/odata/v2/odata-batch-request-builder.ts:254]: I am confused but the limit and [0]. This means the results array contains only on element and this is the one we take, right? I wonder when can the if (!partSeparator) case happen. If the response is defined the split will always return an element.

File: packages/generator/src/edmx-parser/legacy/common/edmx-to-vdm.ts [hunk 2]
```diff
@@ -1,8 +1,7 @@
 /* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
 
-import { last } from 'rambda';
 import { toTitleFormat, toTypeNameFormat } from '@sap-cloud-sdk/core';
-import { createLogger } from '@sap-cloud-sdk/util';
+import { createLogger, last } from '@sap-cloud-sdk/util';
```
Comment (FrankEssenberger) [inline packages/generator/src/edmx-parser/legacy/common/edmx-to-vdm.ts:4]: was this method unused?

File: packages/generator/src/edmx-to-vdm/common/entity.ts [hunk 3]
```diff
@@ -1,6 +1,5 @@
 /* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
-import { createLogger } from '@sap-cloud-sdk/util';
-import { last } from 'rambda';
+import { createLogger, last } from '@sap-cloud-sdk/util';
```
Comment (FrankEssenberger) [inline packages/generator/src/edmx-to-vdm/common/entity.ts:2]: also here.

File: packages/core/src/scp-cf/environment-accessor.ts [hunk 4]
```diff
@@ -41,7 +40,7 @@ export function getDestinationBasicCredentials(): BasicCredentials {
  * @returns The 'destination' credentials object or null if it does not exist.
  */
 export function getDestinationServiceCredentials(): any {
-  return head(getDestinationServiceCredentialsList());
+  return getDestinationServiceCredentialsList()[0];
```
Comment (FrankEssenberger) [inline packages/core/src/scp-cf/environment-accessor.ts:43]: I would prefer to have a `first` or `head` funciton in the util to get the element. I know it is trivial, but if we use it for last I want to symmetric on first.

**Review by FrankEssenberger — CHANGES_REQUESTED** [review]
QUestion

File: packages/core/src/odata/v2/odata-batch-request-builder.ts (hunk 1 above)
Comment (marikaner) [reply packages/core/src/odata/v2/odata-batch-request-builder.ts:254]: Yeah don't look at this. I removed that in my other PR ;) Here I really just replaced last and head.

File: packages/core/src/scp-cf/environment-accessor.ts (hunk 4 above)
Comment (marikaner) [reply packages/core/src/scp-cf/environment-accessor.ts:43]: Hmm ok ;)

File: packages/generator/src/edmx-parser/legacy/common/edmx-to-vdm.ts (hunk 2 above)
Comment (marikaner) [reply packages/generator/src/edmx-parser/legacy/common/edmx-to-vdm.ts:4]: No I added it ;)

File: packages/generator/src/edmx-to-vdm/common/entity.ts (hunk 3 above)
Comment (marikaner) [reply packages/generator/src/edmx-to-vdm/common/entity.ts:2]: Same here ;)

File: packages/generator/src/edmx-parser/legacy/common/edmx-to-vdm.ts (hunk 2 above)
Comment (FrankEssenberger) [reply packages/generator/src/edmx-parser/legacy/common/edmx-to-vdm.ts:4]: But I do not see a usage in this file.

File: packages/generator/src/edmx-to-vdm/common/entity.ts (hunk 3 above)
Comment (FrankEssenberger) [reply packages/generator/src/edmx-to-vdm/common/entity.ts:2]: But I do not see a usage in this file.

File: packages/core/src/odata/v2/odata-batch-request-builder.ts [hunk 5]
```diff
@@ -268,7 +267,7 @@ function partitionChangeSetResponse(
   responseOfSingleRetrieveRequest: string,
   lineBreak: string
 ): string[] {
-  const firstLine = head(responseOfSingleRetrieveRequest.split(lineBreak, 1));
+  const firstLine = responseOfSingleRetrieveRequest.split(lineBreak, 1)[0];
```
Comment (FrankEssenberger) [inline packages/core/src/odata/v2/odata-batch-request-builder.ts:270]: use also first here.

**Review by FrankEssenberger — APPROVED** [review]
Perhaps replace the other [0] also with first. But you mentioned that this is changed in your PR anyhow.

**Pull request merged**